### PR TITLE
HCAP-1510 allow only vancouver island and interior to access MHAW

### PR DESCRIPTION
**Pull request by sunghwan-park-fw**
- Hide disabled location and program filters
- Allow only vancouver island and interior HA and employer users to access MHAW

### CODE REVIEW

File: client/src/pages/private/ParticipantTable.js [hunk 1]
```diff
@@ -379,14 +379,22 @@ const ParticipantTable = () => {
     );
 
     if (isMoH || isHA || isSuperUser) {
-      // should be able to see all users in both MHAW/ HCA programs
-      setPrograms(allPrograms);
+      if (
+        isHA &&
+        !['region_vancouver_island', 'region_interior'].some((region) => roles.includes(region))
```
Comment (fw-noel) [inline client/src/pages/private/ParticipantTable.js:384]: Can you pop this array in a const at the top?

Comment (sunghwan-park-fw) [reply client/src/pages/private/ParticipantTable.js]: Done for BE and FE

File: server/services/participants/participant-entries.ts [hunk 2]
```diff
@@ -188,13 +186,17 @@ export const getParticipants = async (
   isIndigenousFilter?: IsIndigenousFilter,
   programFilter?: ProgramFilter
 ) => {
+  // MHAW program isn't open yet to other regions
+  const hcaOnly =
+    user?.isHA && !MHAW_ENABLED_REGIONS.some((region) => user?.roles.includes(region));
```
Comment (coryatwater-FW) [inline server/services/participants/participant-entries.ts:191]: we should be checking for `isMHSUEmployer` too no?

File: server/services/participants/participant-entries.ts [hunk 3]
```diff
@@ -188,13 +186,17 @@ export const getParticipants = async (
...
+
   // Get user ids
   const participantsFinder = new ParticipantsFinder(dbClient, user);
   const interestFilter =
     isPrivateEmployerOrMHSUEmployerOrHA(user) && statusFilters?.includes('open');
   let participants = await participantsFinder
     .filterRegion(regionFilter)
-    .filterProgram(programFilter)
+    .filterProgram(hcaOnly ? 'HCA' : programFilter)
```
Comment (coryatwater-FW) [inline server/services/participants/participant-entries.ts:199]: need to update the `isMHSUEmployer` check here to use `programFilter` if it exists, currently just hardcoded to `MHAW`

Comment (coryatwater-FW) [inline server/services/participants/participant-entries.ts:199]: nevermind!

File: server/services/participants/participant-entries.ts (hunk 2 above)
Comment (sunghwan-park-fw) [reply server/services/participants/participant-entries.ts:191]: Why? I think the region has a higher priority than the employer type.

Comment (coryatwater-FW) [reply server/services/participants/participant-entries.ts:191]: I don't see anywhere limiting MHSU employer's by interior and costal.  If a MHSU employer from Fraser logs in, these changes won't show participants to them?

**Review by fw-noel — APPROVED** [review]

**Pull request merged**